src/toil_lib/tools/variant_callers.py

```diff
@@ -0,0 +1,381 @@
+import logging
+import os
+import time
+
+from toil.lib.docker import dockerCall
+
+from toil_lib.tools import log_runtime
+
+_log = logging.getLogger(__name__)
+
+def run_freebayes(job, ref, ref_fai, bam, bai, chunksize=None):
+    '''
+    Calls FreeBayes to call variants.
+
+    If job.cores > 1 and chunksize is not None, then runs FreeBayes
+    parallel mode. If not, then runs single threaded FreeBayes.
+
+    :param JobFunctionWrappingJob job: passed automatically by Toil.
+    :param str ref: The reference FASTA FileStoreID.
+    :param str ref_fai: The reference FASTA index FileStoreID.
+    :param str bam: The FileStoreID of the BAM to call.
+    :param str bai: The FileStoreID of the BAM index.
+    :param int chunksize: The size of chunks to split into and call in parallel.
+      Defaults to None.
+    '''
+    work_dir = job.fileStore.getLocalTempDir()
+    file_ids = [ref, ref_fai, bam, bai]
+    file_names = ['ref.fa', 'ref.fa.fai', 'sample.bam', 'sample.bam.bai']
+    for file_store_id, name in zip(file_ids, file_names):
+        job.fileStore.readGlobalFile(file_store_id, os.path.join(work_dir, name))
+
+    output_vcf = os.path.join(work_dir, 'sample.vcf')
+
+    if job.cores > 1 and chunksize:
+        _log.info('Running FreeBayes parallel with %dbp chunks and %d cores',
+                  chunksize,
+                  job.cores)
+        
+        docker_parameters = ['--rm',
+                             '--log-driver', 'none',
+                             '-v', '{}:/data'.format(work_dir),
+                             '--entrypoint=/opt/cgl-docker-lib/freebayes/scripts/fasta_generate_regions.py']
+        start_time = time.time()
+        dockerCall(job,
+                   workDir=work_dir,
+                   dockerParameters=docker_parameters,
+                   parameters=['/data/ref.fa.fai', str(chunksize)],
+                   tool='quay.io/ucsc_cgl/freebayes',
+                   outfile=open(os.path.join(work_dir, 'regions')))
+        end_time = time.time()
+        log_runtime(job, start_time, end_time, 'fasta_generate_regions')
+
+        docker_parameters = ['--rm',
```

Comment:
more DRY to have a `base_docker_parameters` as you're just extending to add `--entrypoint`.